src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollectionOfT.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an <see cref="IEdmObject"/> that is a collection of <see cref="IEdmChangedObject"/>s.
+    /// </summary>
+    [NonValidatingParameterBinding]
+    public class EdmChangedObjectCollection<TStructuralType> : EdmChangedObjectCollection, ICollection<IEdmChangedObject<TStructuralType>>, IEdmObject
+    {
+        private Collection<IEdmChangedObject<TStructuralType>> _items;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmChangedObjectCollection"/> class.
+        /// </summary>
+        /// <param name="entityType">The Edm entity type of the collection.</param>
+        public EdmChangedObjectCollection(IEdmEntityType entityType)
+            : base(entityType)
+        {
+            _items = new Collection<IEdmChangedObject<TStructuralType>>();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmChangedObjectCollection"/> class.
+        /// </summary>
+        /// <param name="entityType">The Edm type of the collection.</param>
+        /// <param name="changedObjectList">The list that is wrapped by the new collection.</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures")]
+        public EdmChangedObjectCollection(IEdmEntityType entityType, Collection<IEdmChangedObject<TStructuralType>> changedObjectList)
+            : base(entityType, changedObjectList as IList<IEdmChangedObject>)
+        {            
+            _items = changedObjectList;
+        }
+
+        /// <inheritdoc/>
+        public override IEnumerable ChangedObjectCollection { get { return _items; } }
+
+        /// <inheritdoc/>
+        public void Add(IEdmChangedObject<TStructuralType> item)
+        {
+            _items.Add(item);
```

Comment:
For each of these methods that affect the membership of the collection, we may also need to call the method on the base, so that if someone casts to the base type they will see the changes.

Please add tests:
1) Create an instance of an EdmChangedObjectCollection<T>, call the various methods to add/remove members, then cast to EdmChangedObjectCollection and verify you see the same members
2) Create an instance of an EdmChangedObjectCollection<T>, cast to EdmChangedObjectCollection, call the various methods to add/remove members, and then verify that you see the changes in the original EdmChangedObjectCollection<T>